File: java/src/main/java/ai/buble/sdk/UnsupportedGenerationFieldException.java
package ai.buble.sdk;

/**
 * Raised when a generation request contains an internal Buble field that is not
 * accepted by the public generation API.
 */
public class UnsupportedGenerationFieldException extends BubleException {
    private final String field;

    public UnsupportedGenerationFieldException(String field) {
        super("Field \"" + field + "\" is an internal Buble field and is not supported by the public generation API.");
        this.field = field;
    }

    public String getField() {
        return field;
    }
}

File: php/src/Exception/UnsupportedGenerationFieldException.php
<?php

declare(strict_types=1);

namespace Buble\Exception;

final class UnsupportedGenerationFieldException extends BubleException
{
    public function __construct(public readonly string $field)
    {
        parent::__construct(sprintf(
            "'%s' is an internal Buble workflow field and cannot be sent to the public generation API.",
            $field,
        ));
    }
}
